### PR TITLE
Add autowiring for HTTPlug

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | https://github.com/symfony/symfony-docs/pull/11820

### CODE REVIEW

**Comment by fabpot** [comment]
Not a big fan of having HttpPlug code in FWB. It should be in a bridge/bundle, but not in the framework which is only about our code.

**Comment by nicolas-grekas** [comment]
I removed the named autowiring aliases because they make no sense actually - nobody should code against HTTPlug anymore.
But since many libs still consume HTTPlug, and since we provide an HttplugClient to ease moving the ecosystem forward, it makes sense to me to have the generic autowiring alias.

I considered moving this to a recipe, but it would need two packages to be applicable (php-http/httplug and symfony/http-client), so that we cannot have one that fits both conditions. Creating a pack would make no sense since the goal of HTTPlug it to fade out now that PSR18 is released.

Moving HttplugClient to the php-http org is possible.
But yet another package :(
While a few lines here make everything so much simpler...

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/http_client.xml [hunk 1]
```diff
@@ -22,5 +22,11 @@
             <argument type="service" id="Psr\Http\Message\StreamFactoryInterface" on-invalid="ignore" />
         </service>
         <service id="Psr\Http\Client\ClientInterface" alias="psr18.http_client" />
+
+        <service id="Http\Client\HttpClient" class="Symfony\Component\HttpClient\HttplugClient">
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/http_client.xml]: not an alias so that it's not listed in debug:autowiring - no need to promote the alias for autowiring

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 2]
```diff
@@ -1901,9 +1906,8 @@ private function registerHttpClientConfiguration(array $config, ContainerBuilder
             $container->registerAliasForArgument($name, HttpClientInterface::class);
 
             if ($hasPsr18) {
-                $container->register('psr18.'.$name, Psr18Client::class)
-                    ->setAutowired(true)
-                    ->setArguments([new Reference($name)]);
+                $container->setDefinition('psr18.'.$name, new ChildDefinition('psr18.http_client'))
+                    ->replaceArgument(0, new Reference($name));
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: this change is not related to httplug - it's just better way to wire the psr18 named aliases

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.
